### PR TITLE
feat: pg17 compatibility

### PR DESCRIPTION
**Pull request by keithf4**
Add compatibility with PG17

 - Restructured background writer metrics to align with changes in PG17. See CHANGELOG for full details

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,21 @@
+2.0.0
+=====
+
+FEATURES
+--------
+ - Compatible with PostgreSQL 17
+
+BREAKING CHANGES
+----------------
+ - PG17 restructured the pg_stat_bgwriter catalog information. Extension has restructured background writer/checkpointer stats around those changes.
```
Comment (hunleyd) [inline CHANGELOG.md:10]:
```suggestion
 - PG17 restructured the pg_stat_bgwriter catalog information. This extension has been restructured around those changes:
```
?

**Review by hunleyd — CHANGES_REQUESTED** [review]

**Review by hunleyd — APPROVED** [review]

File: updates/pgmonitor--1.0.1--2.0.0.sql [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+ELSE
+
+    RETURN QUERY
+    SELECT
+        b.buffers_clean
+        , b.maxwritten_clean
+        , b.buffers_alloc
+    FROM pg_catalog.pg_stat_bgwriter b;
+
+END IF;
```
Comment (pgguru) [inline updates/pgmonitor--1.0.1--2.0.0.sql:34]: Are these queries supposed to be different, or are we supposed to not have the version flag here?

File: sql/functions/functions.sql [hunk 3]
```diff
@@ -628,3 +628,107 @@ END IF;
...
+    FROM pg_catalog.pg_stat_bgwriter b;
+
+ELSE
+
+    RETURN QUERY
+    SELECT
+        b.buffers_clean
+        , b.maxwritten_clean
+        , b.buffers_alloc
+    FROM pg_catalog.pg_stat_bgwriter b;
```
Comment (pgguru) [inline sql/functions/functions.sql:659]: Are these supposed to be the same query?

Comment (keithf4) [reply sql/functions/functions.sql:659]: Hah! This was left over from my initial writeup of this where I only had one metric for everything. I'd slowly moved the pieces from this one to the other new ones and of course it left these equivalent in the end.

Comment (keithf4) [reply sql/functions/functions.sql]: Thanks for that catch. Removed the function for that and just made it a normal view

File: updates/pgmonitor--1.0.1--2.0.0.sql (hunk 2 above)
Comment (keithf4) [reply updates/pgmonitor--1.0.1--2.0.0.sql]: Same as other catch

**Review by hunleyd — APPROVED** [review]

**Review by pgguru — APPROVED** [review]

**Pull request merged**